### PR TITLE
New event "adminhtml_sales_order_create_save_before" when editing an order.

### PR DESCRIPTION
**Pull request by woutersamaey**
New event for when you're editing an order in Magento Admin and need to make changes to the old and/or new order before save.

### CODE REVIEW

File: app/code/core/Mage/Adminhtml/Model/Sales/Order/Create.php [hunk 1]
```diff
@@ -1565,6 +1565,9 @@ public function createOrder()
         if ($oldOrder->getId()) {
             $oldOrder->setRelationChildId($order->getId());
             $oldOrder->setRelationChildRealId($order->getIncrementId());
+
+            Mage::dispatchEvent('order_edited_save_before', ['new_order' => $order, 'old_order' => $oldOrder]);
```
Comment (kiatng) [inline app/code/core/Mage/Adminhtml/Model/Sales/Order/Create.php:1569]: How about event name `adminhtml_sales_order_create_save_before`?

Comment (woutersamaey) [reply app/code/core/Mage/Adminhtml/Model/Sales/Order/Create.php]: Done

**Review by kiatng — APPROVED** [review]
LGTM

**Comment by woutersamaey** [comment]
What does LGTM mean?

**Review by sreichel — COMMENTED** [review]
lgtm

Q: short array syntax y/n?

**Comment by colinmollenhour** [comment]
> lgtm
> 
> Q: short array syntax y/n?

Short array syntax is fine since PHP 5.4 and we list PHP 7.0 as a requirement (although I think 5.6 is still working).

**Review by sreichel — APPROVED** [review]

**Pull request merged**